pytest-{{cookiecutter.plugin_name}}/.github/workflows/main.yml

```
@@ -0,0 +1,40 @@
+name: main
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        include:
+          - python-version: '3.8'
+            tox-env: 'py38'
+          - python-version: '3.9'
+            tox-env: 'py39'
+          - python-version: '3.10'
+            tox-env: 'py310'
+          - python-version: '3.11'
+            tox-env: 'py311'
+          - python-version: '3.12'
+            tox-env: 'py312'
+          - python-version: 'pypy-3.8'
+            tox-env: 'pypy3'
+    steps:
+      - uses: actions/checkout@v4
+
+      - uses: actions/setup-python@v5
+        with:
+          python-version: {% raw -%}${{ matrix.python-version }}{% endraw %}
+
+      - name: Install tox
+        run: pip install tox
+
+      - name: Test
+        run: tox -e {% raw -%}${{ matrix.tox-env }}{% endraw -%}
```

`tox -e py` runs using the current interpreter, so this can be simplified:

```suggestion
jobs:
  test:
    runs-on: ubuntu-latest
    strategy:
      matrix:
        python-version: ['3.8', '3.9', '3.10', '3.11', '3.12', 'pypy-3.8']
    steps:
      - uses: actions/checkout@v4

      - uses: actions/setup-python@v5
        with:
          python-version: {% raw -%}${{ matrix.python-version }}{% endraw %}

      - name: Install tox
        run: pip install tox

      - name: Test
        run: tox -e py
```

---

I never seen that `raw` directive in `python-version`, what is that for?

---

Updated, thanks.

The `raw` is escaping for the Jinja template language which cookiecutter uses, otherwise the `${{ .. }}` is interpreted by Jinja.